addons/l10n_id_efaktur/models/account_move.py

```diff
@@ -56,10 +56,18 @@ def _compute_kode_transaksi(self):
         for move in self:
             move.l10n_id_kode_transaksi = move.partner_id.l10n_id_kode_transaksi
 
-    @api.depends('partner_id')
+    @api.depends('partner_id', 'line_ids.tax_ids')
     def _compute_need_kode_transaksi(self):
         for move in self:
-            move.l10n_id_need_kode_transaksi = move.partner_id.l10n_id_pkp and not move.l10n_id_tax_number and move.move_type == 'out_invoice' and move.country_code == 'ID'
+            # If there are no taxes at all on every line (0% taxes counts as having a tax) then we don't need a kode transaksi
+            move_has_taxes = any(line.tax_ids for line in move.line_ids if line.display_type == 'product')
+            move.l10n_id_need_kode_transaksi = (
+                    move.partner_id.l10n_id_pkp
+                    and not move.l10n_id_tax_number
+                    and move.move_type == 'out_invoice'
+                    and move.country_code == 'ID'
+                    and move_has_taxes
```

Comment:
Perhaps a bit too indented ? O:)

Reply:
Hmm according to Pycharm auto-formatting, it's correct with two tabs.
But indeed it's a bit weird, I changed it to one tab.